src/main/daml/Daml/Finance/Instrument/Bond/Callable/Factory.daml

```diff
@@ -55,7 +56,10 @@ template Factory
               lastEventTimestamp
               prevEvents
               observers
+          cid <- toInterfaceContractId <$> create callableInstrument
           BaseInstrument.createReference instrument.depository $ toInterfaceContractId cid
+          Claim.getClaims (toInterface @Claim.I callableInstrument) $
```

Comment:
This creates a hidden dependency of the bond factory on the `Calendar` implementation (via keys). It might be fine after all, but I was wondering if you tried just validating the schedule?